File: go/patch.go
package main

import (
	"fmt"
	"os"
	"time"
)

func patchFile(filename string, bytes []byte, offset int64) error {
	file, err := os.OpenFile(filename, os.O_RDWR, 0644)
	if err != nil {
		return err
	}
	defer file.Close()

	_, err = file.Seek(offset, 0)
	if err != nil {
		return err
	}

	_, err = file.Write(bytes)
	return err
}

func main() {
	start := time.Now()

	filename := "randomblock"
	bytes := []byte("d1ffd1ff")
	offset := int64(100)

	err := patchFile(filename, bytes, offset)
	if err != nil {
		panic(err)
	}

	duration := time.Since(start)
	fmt.Printf("Time taken: %v (go)\n", duration)
}

File: rust/patch.rs
use std::fs::OpenOptions;
use std::io::{Seek, SeekFrom, Write};
use std::time::Instant;

fn patch_file(filename: &str, bytes: &[u8], offset: u64) -> std::io::Result<()> {
    let mut file = OpenOptions::new().read(true).write(true).open(filename)?;
    file.seek(SeekFrom::Start(offset))?;
    file.write_all(bytes)?;
    Ok(())
}

fn main() {
    let start = Instant::now();
    let filename = "randomblock";
    let bytes = b"d1ffd1ff";
    let offset = 100;
    
    if let Err(e) = patch_file(filename, bytes, offset) {
        eprintln!("Error: {}", e);
    }

    let duration = start.elapsed();
    println!("Time taken: {:?} (rust)", duration);
}